src/Compilers/CSharp/Portable/Symbols/SubstitutedTypeParameterSymbol.cs

```diff
@@ -92,9 +92,14 @@ public override string Name
             }
         }
 
-        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        public sealed override ImmutableArray<CSharpAttributeData> GetAttributes()
         {
-            return _underlyingTypeParameter.GetAttributes();
+            if (_container is SynthesizedMethodBaseSymbol { InheritsBaseMethodAttributes: true })
+            {
+                return _underlyingTypeParameter.GetAttributes();
+            }
+
+            return ImmutableArray<CSharpAttributeData>.Empty;
```

Comment:
>return ImmutableArray<CSharpAttributeData>.Empty; [](start = 12, length = 49)

This doesn't feel right. A type parameter of a generic type nested into a constructed type should return attributes from the original definition of the type parameter (please add a test). I think this logic should be moved to ```SynthesizedSubstitutedTypeParameterSymbol```. Please check if VB has the same problem around emitting attributes when not supposed to.

Reply:
I ended up moving the GetAttributes() to the WrappedTypeParameterSymbol, it seems to better resemble what WrappedParameterSymbol does, and it felt like a similar default behavior was appropriate.

I added the test `SubstitutedTypeParameter_Attributes` which demonstrates that we carry over the type parameter attributes from the declaration after constructing the containing type.

I added a VB test which shows that the base method and return attributes are not inherited on a synthesized method. Type parameter attributes seem a little trickier because type parameter attributes don't seem to be available in VB. I could maybe try having the VB program override a method with a type parameter attribute which is defined somewhere else.

---
In reply to: [346603239](https://github.com/dotnet/roslyn/pull/39830#discussion_r346603239) [](ancestors = 346603239)

Reply:
> I could maybe try having the VB program override a method with a type parameter attribute which is defined somewhere else.

Yes please define the base class in C#. We have a helper called CreateCSharpCompilation (or similar) that VB tests use for scenarios like this. It should be possible to find examples.


---
In reply to: [346624109](https://github.com/dotnet/roslyn/pull/39830#discussion_r346624109) [](ancestors = 346624109,346603239)

Reply:
Added the test in d933f55.

---
In reply to: [346949985](https://github.com/dotnet/roslyn/pull/39830#discussion_r346949985) [](ancestors = 346949985,346624109,346603239)